File: VleisurePartner.Web/Controllers/HomeController.cs
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using VleisurePartner.Domain;
using VleisurePartner.EF;
using VleisurePartner.Web.Models;
using VleisurePartner.Web.Models.RequestModels;
using VleisurePartner.Web.Infrastructure;
using VleisurePartner.Web.Models.ResponseModels;
using VleisurePartner.Web.Services;

namespace VleisurePartner.Web.Controllers
{
    public class HomeController : Controller
    {
        private readonly IVleisureApiRequest _vleisureApiRequest;
        private readonly IContext _appDbContext;

        public HomeController(IVleisureApiRequest vleisureApiRequest,
            IContext appDbContext)
        {
            _vleisureApiRequest = vleisureApiRequest;
            _appDbContext = appDbContext;
        }

        public ActionResult Index()
        {

            return View();
        }

        [HttpPost]
        public ProxyResult<HotelListRs> GetHotelList(HotelListRequest req)
        {
        //    var requestBody = new HotelListRequest
        //    {
        //        CityCode = "",
        //        ArrivalDate = "12/29/2018",
        //        DepartureDate = "12/30/2018",
        //        RoomGuests = new List<RoomGuestRequestModel>(),
        //        HotelIds = new int[]{
        //        632882,
        //        148036,
        //        100502 }
        //    };
        //    var roomGuest = new RoomGuestRequestModel();
        //    roomGuest.NumberOfAdults = 1;
        //    requestBody.RoomGuests.Add(roomGuest);
            
            var operationResult = _vleisureApiRequest.GetHotelList(req);


            return operationResult.ToProxyResult();
        }
        
        [HttpPost]
        public ProxyResult<HotelDetailsResponse> GetHotelDetails(HotelDetailsRequest req)
        {
            var requestBody = new HotelDetailsRequest
            {
                ArrivalDate = "12/29/2018",
                DepartureDate = "12/30/2018",
                RoomGuests = new List<RoomGuestRequestModel>(),
                HotelId = 632882
            };
            var roomGuest = new RoomGuestRequestModel();
            roomGuest.NumberOfAdults = 1;
            requestBody.RoomGuests.Add(roomGuest);
            
            var operationResult = _vleisureApiRequest.GetHotelDetails(requestBody);

            return operationResult.ToProxyResult();
        }

        [HttpPost]
        public ProxyResult<RoomAvailabilityResponse> GetRoomAvailability(RoomAvailabilityRequest req)
        {
            var operationResult = _vleisureApiRequest.GetRoomAvailability(req);


            return operationResult.ToProxyResult();
        }

        [HttpPost]
        public ProxyResult<IEnumerable<RegionViewModel>> GetRegions(string searchString)
        {
            var regions = _appDbContext.Regions
                .Where(x => x.RegionNameLong.Contains(searchString))
                .ToList()
                .Select(x =>
                
                    new RegionViewModel
                    {
                        RegionId = x.RegionId,
                        RegionNameLong = x.RegionNameLong
                    }
                )
                .OrderBy(x => x.RegionNameLong.Length)
                .ToList();

            return ProxyResult<IEnumerable<RegionViewModel>>.Success(regions);
        }
        
    }
}

File: VleisurePartner.Web/src/proxy/HomeController.ts


/* tslint:disable */

import { HttpService } from '@infrastructure/transport/httpservice'
import OperationResult from '@infrastructure/OperationResult'
import { AppConfig } from '@infrastructure/config/config'


export class HomeController {
    
        public index = (): Promise<OperationResult<any>> => {
            const route = `api/Home/`;
            return HttpService.Instance.request({
                url: this.GetUrl(route, 'HomeController','Index'),
                method: "post",
                data: null,
                headers: {'X-Requested-With': 'XMLHttpRequest'}
            });
        };
    
        public getHotelList = (req: ProxyModel.HotelListRequest): Promise<OperationResult<ProxyModel.HotelListRs>> => {
            const route = `api/Home/`;
            return HttpService.Instance.request({
                url: this.GetUrl(route, 'HomeController','GetHotelList'),
                method: "post",
                data: req,
                headers: {'X-Requested-With': 'XMLHttpRequest'}
            });
        };
    
        public getHotelDetails = (req: ProxyModel.HotelDetailsRequest): Promise<OperationResult<ProxyModel.HotelDetailsResponse>> => {
            const route = `api/Home/`;
            return HttpService.Instance.request({
                url: this.GetUrl(route, 'HomeController','GetHotelDetails'),
                method: "post",
                data: req,
                headers: {'X-Requested-With': 'XMLHttpRequest'}
            });
        };
    
        public getRoomAvailability = (req: ProxyModel.RoomAvailabilityRequest): Promise<OperationResult<ProxyModel.RoomAvailabilityResponse>> => {
            const route = `api/Home/`;
            return HttpService.Instance.request({
                url: this.GetUrl(route, 'HomeController','GetRoomAvailability'),
                method: "post",
                data: req,
                headers: {'X-Requested-With': 'XMLHttpRequest'}
            });
        };
    
        public getRegions = (searchString: string): Promise<OperationResult<ProxyModel.RegionViewModel[]>> => {
            const route = `api/Home/?searchString=${encodeURIComponent(searchString)}`;
            return HttpService.Instance.request({
                url: this.GetUrl(route, 'HomeController','GetRegions'),
                method: "post",
                data: null,
                headers: {'X-Requested-With': 'XMLHttpRequest'}
            });
        };
    
    
    
    private GetUrl(route: string, controller : string,  action : string): string {
        controller = controller.replace('Controller','');

        const param : string[] = route.split("?");
        if(param[1]){
            return `${controller}/${action}?${param[1]}`;
        }

        return `${controller}/${action}`;
    }
}
